lib/python/rosie/ws.py

```diff
@@ -61,35 +77,90 @@ def __init__(self, *args, **kwargs):
                 ResourceLocator.default().get_util_home(
                     "lib", "html", "template", "rosie-disco")))
         db_url_map = {}
-        for key, node in rose_conf.get(["rosie-db"]).value.items():
+        for key, node in list(rose_conf.get(["rosie-db"]).value.items()):
             if key.startswith("db.") and key[3:]:
                 db_url_map[key[3:]] = node.value
         self.db_url_map = db_url_map
         if not self.db_url_map:
```

Comment:
Another random question - given the line above, is this ever true?

Reply:
Good question, & well spotted. I can't say I have combed through the original code but it would be natural that it can be improved in ways such as this. Having though about the logic here, I agree that there seems to be no need for the ``if not`` statement & contained assignment, so I will remove it.